### PR TITLE
Move PluginsService to its own internal package

### PR DESCRIPTION
**Pull request by prdoyle**
Move `PluginsService` and associated machinery into an `internal` package called `org.elasticsearch.plugins.internal.service`.

As discussed at yesterday's tech debt meeting, this would allow us to consider adding methods that we'd be unwilling to make publicly available to the rest of the code.  cc: @rjernst 

Classes moved:

```
ModuleSupport.java
PluginBundle.java
PluginIntrospector.java
PluginLoaderIndirection.java
PluginsService.java
PluginsUtils.java
StablePluginPlaceHolder.java
UberModuleClassLoader.java
```

### CODE REVIEW

**Comment by prdoyle** [comment]
HOMER does not like my PR.

**Comment by prdoyle** [comment]
Turns out this depends on #109653.

**Review by rjernst — COMMENTED** [review]
I assume this was completely mechanical (just refactor tools in IntelliJ)?

I have a slight preference personally for flatter package names, so i would put this under `o.e.plugins.service` or perhaps more abstractly `o.e.plugins.loading` (or loader/load/something?) since it is not a "service" like other services in ES.

**Comment by prdoyle** [comment]
Mostly mechanical, plus a couple of things IntelliJ missed, like [this](https://github.com/elastic/elasticsearch/pull/109872/files#diff-3fae7b40965cf77de40588cd5af5ddde50af6483168dc687d73e006fb4536957L23-R23).

Sure, I can change it to `loading`. I misunderstood and thought you wanted it under `internal`.

**Comment by prdoyle** [comment]
(Side note: seems tragic that checkstyle needs to use backslashes for file paths. Java on windows can use forward slashes just fine.)

File: server/src/main/java/module-info.java [hunk 1]
```diff
@@ -460,4 +460,5 @@
             org.elasticsearch.serverless.shardhealth,
             org.elasticsearch.serverless.apifiltering;
     exports org.elasticsearch.lucene.spatial;
+    exports org.elasticsearch.plugins.loading;
```
Comment (prdoyle) [inline server/src/main/java/module-info.java:463]: Well this is ironic.

Comment (prdoyle) [reply server/src/main/java/module-info.java:463]: Looks like I was cheating. The IDE automatically exported the new package, making me miss a bunch of cross-module access problems. I'll sort that out and move the PR out of draft when it's ready again.

**Comment by prdoyle** [comment]
@rjernst - how do you feel about putting plugin-cli in a Java module so we can export our package to it? See [here](https://github.com/elastic/elasticsearch/pull/109872/commits/7ebb41048e637da0d7a90c7675b9567cb96b6a14).

**Review by rjernst — APPROVED** [review]
LGTM

**Pull request merged**